### PR TITLE
[FS-172266, FS-199069] : Conductor Monitor Changes for queue depth metrics

### PR DESCRIPTION
**Pull request by keerthivaasan-kanagaraj**
Pull Request type
----
- [ ] Bugfix
- [ ] Feature
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes (Please run `./gradlew generateLock saveLock` to refresh dependencies)
- [ ] WHOSUSING.md
- [ ] Other (please describe):

**NOTE**: Please remember to run `./gradlew spotlessApply` to fix any format violations.

Changes in this PR
----

_Describe the new behavior from this PR, and why it's needed_
Issue #

Alternatives considered
----

_Describe alternative implementation you have considered_

### CODE REVIEW

File: server/src/main/java/com/netflix/conductor/config/PrometheusIntegrationConfig.java [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+
+    @Override
+    public void run(String... args) throws Exception {
+        setupPrometheusRegistry();
+    }
+
+    /**
+     * To Register PrometheusRegistry
+    */
+    private static void setupPrometheusRegistry() {
```
Comment (ab48917) [inline server/src/main/java/com/netflix/conductor/config/PrometheusIntegrationConfig.java]: is this working Keerthi? 
Can you create a separate PR for this alone?

File: core/src/main/java/com/netflix/conductor/core/reconciliation/WorkflowSweeper.java [hunk 2]
```diff
@@ -100,10 +100,13 @@ public void sweep(String workflowId) {
...
-            long startTime = Instant.now().toEpochMilli();
+            // long startTime = Instant.now().toEpochMilli();
             unack(workflow, workflowOffsetTimeout);
-            long endTime = Instant.now().toEpochMilli();
-            Monitors.recordUnackTime(workflow.getWorkflowName(), endTime - startTime);
+            // long endTime = Instant.now().toEpochMilli();
+            // This metrics(recordUnackTime) has workflowName in our prod account we have more number of workflows.
+            // So we may get cardinality problems in haystack due to this.
+            // Disabling this temporarily.
+            // Monitors.recordUnackTime(workflow.getWorkflowName(), endTime - startTime);
```
Comment (geethalladi) [inline core/src/main/java/com/netflix/conductor/core/reconciliation/WorkflowSweeper.java]: Can we replace this with a log entry ?

File: core/src/main/java/com/netflix/conductor/metrics/WorkflowMonitor.java [hunk 3]
```diff
@@ -74,51 +74,60 @@ public WorkflowMonitor(
...
-                                long count =
-                                        executionDAOFacade.getPendingWorkflowCount(workflowName);
-                                Monitors.recordRunningWorkflows(count, workflowName, ownerApp);
-                            });
-
-            taskDefs.forEach(
-                    taskDef -> {
+        }
+
+        // Commented out as we don't have use case for Pending Workflows as of now and hence we don't have pending workflow implementation from
```
Comment (geethalladi) [inline core/src/main/java/com/netflix/conductor/metrics/WorkflowMonitor.java]: - Remove these code if they are no longer needed
- We can always get them from the version control

File: server/src/main/java/com/netflix/conductor/config/PrometheusIntegrationConfig.java [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+import io.micrometer.core.instrument.Clock;
+import io.micrometer.core.instrument.Metrics;
+
+// This class loads all the configurations related to prometheus. 
+@Configuration
+public class PrometheusIntegrationConfig
+        implements CommandLineRunner {
+
+    private static final Logger log = LoggerFactory.getLogger(PrometheusIntegrationConfig.class);
+    private static PrometheusMeterRegistry prometheusRegistry;
```
Comment (geethalladi) [inline server/src/main/java/com/netflix/conductor/config/PrometheusIntegrationConfig.java]: - Make this field and `setupPrometheusRegistry` as non-static instance properties.
- No specific reason for them to be static

**Review by geethalladi — DISMISSED** [review]
Mostly good. +1

File: core/src/main/java/com/netflix/conductor/core/reconciliation/WorkflowSweeper.java (hunk 2 above)
Comment (keerthivaasan-kanagaraj) [reply core/src/main/java/com/netflix/conductor/core/reconciliation/WorkflowSweeper.java]: we need to add bb logs for sweeper and reconciler with logs standardisation task planning to add that log

ref: https://docs.google.com/spreadsheets/d/1GVa5Vwk_NW6fY5y9MtnaH0nIv06VcynldO87CNFXjqs/edit?gid=0#gid=0

**Review by prabhakaranE6157 — DISMISSED** [review]

**Review by geethalladi — APPROVED** [review]
+1 LGTM

**Pull request merged**